processor/cumulativetodeltaprocessor/internal/tracking/tracker.go

```diff
@@ -81,14 +123,42 @@ func (t *MetricTracker) Convert(in MetricPoint) (out DeltaValue, valid bool) {
 	hashableID := b.String()
 	identityBufferPool.Put(b)
 
+	var keep bool
 	s, ok := t.states.LoadOrStore(hashableID, &State{
 		PrevPoint: metricPoint,
 	})
 	if !ok {
-		return
+		switch t.initialValue {
+		case InitialValueDrop:
+			return
+		case InitialValueAuto:
+			if metricID.StartTimestamp < t.startTime || metricPoint.ObservedTimestamp < metricID.StartTimestamp {
+				return
+			}
+			out.StartTimestamp = metricID.StartTimestamp
+			keep = true
+		case InitialValueKeep:
+			out.StartTimestamp = metricID.StartTimestamp
+			if metricID.StartTimestamp == 0 {
+				out.StartTimestamp = t.startTime
+			}
```

Comment:
I'm a bit unsure about the correct value for start time if we unconditionally keep the value

Reply:
alright, i've removed the timestamp